moveos/moveos-types/src/move_types.rs

```diff
@@ -369,7 +374,11 @@ pub fn parse_type_tag(s: &str) -> Result<TypeTag> {
 pub fn parse_struct_tag(s: &str) -> Result<StructTag> {
     let s = s.trim();
     if s.starts_with("0x") {
-        StructTag::from_str(s)
+        if let Ok(struct_tag) = StructTag::from_str(s) {
+            Ok(struct_tag)
+        } else {
+            struct_tag_from_canonical_string(s)
```

Comment:
We can directly call `StructTag::from_str` now and remove the struct_tag_from_canonical_string.

Reply:
done